source/objects/bodyClass.m

```diff
@@ -781,92 +786,116 @@ function qtfExcitation(obj, wavesObj, iH)
+            QTF.sumExtended(end,:,:) = QTF.sumExtended(end-1,:,:);
 
-            Omega_coarse = [Omega_fine(1) Omega_coarse Omega_fine(end)];
-            nOmega = length(Omega_fine);
+            omegaCoarse = [omegaFine(1) omegaCoarse max(omegaFineExtended)];
 
-            if obj.QTFs == 1 % Calculates full difference QTFs
+            nOmega = length(omegaFine);
+
+            if obj.QTFs > 0          
                 for n = 1 : obj.dof
-                    % Slowly varing component calculation
-                    QTF.Diff_refined(:,:,n) = griddata(Omega_coarse, Omega_coarse, QTF.DiffExtended(:,:,n), Omega_x, Omega_y);
-                    tmp =  real(sum(Amp_freq .* conj(Amp_freq) .* diag(QTF.Diff_refined(:,:,n))));
-                    fMeanDriftLoad = 2/(N)^2 * sum(tmp)*ones(N,1) * 2; % Multiplied by 2 for the two-sided spectrum
-                    Hu = zeros(N,1);
-                    for nu = 1 : nOmega - 1
-                        for l = 1 : nOmega- nu
-                            Hu(nu+1) = Hu(nu+1) + 2*(Amp_freq(l+nu) .* conj(Amp_freq(l)) .* QTF.Diff_refined(l+nu,l,n));
+                    %% Slowly varing component calculation
+                    if obj.QTFs == 1   % Full QTFs calculation
+                        QTF.diffRefined(:,:,n) = griddata(omegaCoarse, omegaCoarse, QTF.diffExtended(:,:,n), omegaX, omegaY);
+                        tmp =  real(sum(ampFreq .* conj(ampFreq) .* diag(QTF.diffRefined(:,:,n))));
+                        fMeanDriftLoad = 2/(N)^2 * sum(tmp)*ones(N,1) * 2;      % Multiplied by 2 for the two-sided spectrum
+                        Hu = zeros(N,1);
+                        for nu = 1 : nOmega - 1
+                            for l = 1 : nOmega- nu
+                                Hu(nu+1) = Hu(nu+1) + 2*(ampFreq(l+nu) .* conj(ampFreq(l)) .* QTF.diffRefined(l+nu,l,n));
+                            end
                         end
+                        Hu(N/2+1:end) = conj(Hu(N/2:-1:1))';
+                        fSlowDriftLoad = 2 * ifft(Hu/N,'symmetric');
+                        obj.hydroForce.(hfName).QTF.fSlowVaryingForces(:,n) = fSlowDriftLoad(1: timeIdx) + fMeanDriftLoad(1: timeIdx);
+
+                    elseif obj.QTFs == 2 % Newman QTFs calculation
+
+                        QTF.diffDiag = interp1(omegaCoarse, real(diag(QTF.diffExtended(:,:,n), 0)), omegaFineExtended)';
+
+                        % Sanity check: match sizes before assigning
+                        assert(isequal(size(ampFreqExtended), size(QTF.diffDiag)), 'QTF: Size mismatch in Newman Approx. lines');
+
+                        Lp = zeros(N,1);
+                        Ln = zeros(N,1);
+                        Lp(1:N/2) = ampFreqExtended(1:N/2) .* sqrt( max(QTF.diffDiag(1:N/2), 0));
+                        Ln(1:N/2) = ampFreqExtended(1:N/2) .* sqrt(-min(QTF.diffDiag(1:N/2), 0));
+
+                        % Compute slow-varying forces
+                        fSlowNewman = abs(2*ifft(Lp)).^2 - abs(2*ifft(Ln)).^2;
```

Comment:
Should this also include a term that is the order of magnitude of the frequency difference?